tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -42,7 +53,12 @@ void _SysFatalErrorHandler(unsigned int reason, const NANO_ESF *pEsf)
 		give_uthread_end_sem = false;
 		k_sem_give(&uthread_end_sem);
 	}
-	ztest_test_pass();
+	if (expect_fault) {
+		expect_fault = false;
+		BARRIER();
+		ztest_test_pass();
+	}
```

Comment:
For ARM platforms, we just need to return here right after the ztest_test_pass.  The k_thread_abort will issue a pendsv to swap.  The swap won't happen until we return from exception, so we'll actually fall through to the zassert_unreachable.

Reply:
no negative, if it is doing that it is a bug.
immediately after we return from the exception, the swap should occur. it should not be executing anything in the application!
See tests/kernel/fatal

Reply:
Ah, good point.  We can either return after ztest_test_pass() or move the zassert_unreachable() inside an else clause.